### ISSUE TITLE
Improving autoland after failsafe-RTH for fixedwing aircraft

### ISSUE THREAD

**Issue by Flight-instructor** [iNavFlight/inav#6438]
Current Behavior
As of now, and as far as I understood the function land after RTH is done by cyrcling around while descendinding until touching the ground. Disadvantage of this kind to execute an attempt to land  makes it impossible to predict the direction of the aircraft when coming closed to the ground making it possible to hit obstacles as collegues or, in most cases worse:-)), the flying-field building. To be honest, coming from the "real" aircraft flying, I prefer a landing straight on runway heading.

Desired Behavior
Well, I thought about a resonable solution, as because I am not coming out of the programmers people, I only can make suggestions that I think are possible to program. 
Lets imagime the following:
Given parameters:
We have the safehome now, that might be airfield middle. 
The airfield has a direction that normally does not change ( as nobody turns it during the night )
Normally, we do not know the exact direction.
And also normally, we try more or less taking of against the wind, so pending on it, either in the one, or the other direction.

Wanted parameters:
Aircraft learns, in what direction and out of what altitude and position a straight landing shall be performed ( or at least a straight approach ).

Suggested Solution
Our aircraft is standing at the end on the runway runway for takeoff  and armed. We put it int a new to create " learn your field" mode. It now takes the apropriate safehome-point as its rth position. We let the aircraft taxi exactly along the runway middle line, or, in case of an aircraft to be hand-launched, we take it in the hand and move with it along that line. GPS now gives a direction that is our runway direction for the used safehome-point.
At this point, INAV should calculate a point 1, which is exactly at a given distance from safehomepoint towards the takeof-line and a point 2 exactly at a given distance backwards.
This will be our final approach-fixes for autoland.
When launched or in takeoff roll, the FC knows the TO direction and chooses the point that is laying backwards as is final approach-fix. Thats all for takeoff. 
After RTH, b.E. in RTH out of failsafe, the aircraft could make a rectangle flightpath towards the final approach fix. Which one? Well, as far as we start our takeoff roll against the wind,  INAV now choses the point, that is backwards to be able to land from that point at that day as its final approach fix. The rectangle could be intercepted and should be big enough, to meet the requirings out of max bank and resulting turn radius. Easy to calculate so far.
Whats happening now? RTH brings the aircraft to the safehome point at a given altitude. Than it starts to fly the aircraft in runway ( takeoff!! ) direction for lets say 20s**. Turn left by 90°, hold dircection for 20 ( user defined )seconds, turning left by 90° and hold TRACK, not heading!!. Our final approach-fix than passes from front left to left and to backwards left. On left, add another 20 seconds, turn left by 150° and intercept your final approach bearing. Between the FAF ( final approach fix and safe home, there is now a nav-line( like a mission-segment ) the aircraft follows, slightly descending to your airfield being safe of all abstacles and with a given speed or at least a given throtle value and with a great chance to land without major demage.
As INAV is able to calculate a cruise descent ( b.e. RTH with low descent rates to arrive RTH-Alt at the RTH point ) it should be able to calculate the descent rate for the final approach also. Aslo a given speed shold be fixed and maintained by FC.

Pending on what direction you took off, the aircraft ( better INAV ) knows which of two stored points is the todays FAF.
I think, when reading this for the first time, it might sound complicated. But I am sure, it is easy to program if enough spare memory is available. 

This only works perfect if approach path is obstacle free but in most model airfields this is given. 
It could be interesting zu add an offset angle for the final to maintain obstacle clearence, but need experienced users.

** this is to bring the aircraft in a defined direction before starting the landing procedure. You never know, a what point RTH-failsafe starts and so we do not know, from which direction the aircraft comes back to field. From here on, there are different ways to solve the problem with our needed flightpath towards the FAF. It could also consist of a couple of procedure-turns as ( fullsized aircraft´s ) autopilots do when approaching autmatically. Well, thats not really complicated.

Nevertheless, this would be an advanced-user feature as the distance of the FAF´s and the altitude needed and so an, is to be calculated by the the user, and therefore, there must be possibillities to to put those parameters in the FC with the configurator.
And while speaking now about the configurator, it would be a pleasure to put in the in a graphic desing ( well wishes, I know!! )

Who does this impact? Who is this for?
Advanced fixwing users with some experience and theoretical knowledge? 

Additional context
Please regard, that I am familiar with flying ( models as well as full size, RC-model flying since 52 years now, and anothother 29 years as a flight instructor for full size  ), but I am new into INAV and FPV flying. Some experience with arkbird some years ago, but now back with matek and INAV. 
btw, I really love the videos from Pawel, as he explains clearly in a perfect english and it is fun to listen.

Is there anything I am unhappy with?
Yes, its the wording in the documetation, but I com back with another threat, cause it not regarding autoland, though when reading this section as well as emergency landing, I did not understand something due to the wording used.

**Comment by MoriXTV** [comment]
Did this get Implemented yet? Just came across it and I must say the Idea is really cool.

Maybe a simpler implementation with just two points like P1: "land here" and P2: "approach from here" that can be set by walking there with the model and hitting a switch would do.
From these two GPS points and the already existing vertical landing velocity, cruise speed and RTH altitude the approach vector and approach point elevation should be easy enough to calculate.

Then if the landing mode is activated (after RTH or via switch) the pane could just fly to the calculated approach point at RTH altitude and begin to descend.

It would be really nice to have this and not having to worry about landing especially for something like the AR-Pro with 4s6p  18650 because we all know that landings with small and heavy planes are either fast and tricky or short and expensive ;)

Love your work guys keep that up.

**Comment by MrD-RC** [comment]
A much improved autoland is currently scheduled for 7.1.
